### PR TITLE
Add Boxel::Field and Boxel::Input components

### PR DESCRIPTION
**Pull request by burieberry**
field

input

### CODE REVIEW

File: addon/components/boxel/input/index.css [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  outline-color: var(--boxel-highlight);
+}
+
+.boxel-input:disabled {
+  background-color: var(--boxel-light-100);
+  border-color: var(--boxel-purple-300);
+  color: var(--boxel-purple-300);
+}
+
+/* TODO: required and disabled styles */
```
Comment (lukemelia) [inline addon/components/boxel/input/index.css:28]: What's the status of these?

File: addon/components/boxel/input/index.css [hunk 2]
```diff
@@ -0,0 +1,28 @@
+.boxel-input {
+  --boxel-input-height: 2.5rem; /* 40px */
+
+  width: 100%;
+  min-height: var(--boxel-input-height);
+  padding: var(--boxel-sp-xxs) 0 var(--boxel-sp-xxs) var(--boxel-sp);
+  border: 1px solid var(--boxel-purple-300);
+  border-radius: 5px;
```
Comment (lukemelia) [inline addon/components/boxel/input/index.css]: Consider using ` --boxel-border-radius-sm`

File: addon/components/boxel/input/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class Input extends Component {
+  @tracked value;
```
Comment (lukemelia) [inline addon/components/boxel/input/index.js]: It doesn't look like this `value` is used by anything. The template using `@value`

File: package.json [hunk 4]
```diff
@@ -40,6 +40,7 @@
     "ember-cli-string-helpers": "^5.0.0",
     "ember-composable-helpers": "^4.4.1",
     "ember-concurrency-decorators": "^2.0.3",
+    "ember-css-element-queries": "^1.0.0",
```
Comment (lukemelia) [inline package.json]: This addon's repository has been archived (https://github.com/simplabs/ember-css-element-queries). Should we consider a different implementation?

**Review by lukemelia — CHANGES_REQUESTED** [review]
General approach looks great. Left a few questions/suggestions.

File: package.json (hunk 4 above)
Comment (burieberry) [reply package.json]: Yeah, I wonder why they archived it. Maybe `css-element-queries` (https://github.com/marcj/css-element-queries) now works for ember out-of-the-box. I was planning to try but forgot about it.

Comment (burieberry) [reply package.json]: Update on this: It didn't work out-of-the-box. There's this note on the readme (https://github.com/marcj/css-element-queries#module-loader), but I'm not sure where it needs to be added.

**Comment by burieberry** [comment]
@lukemelia By the way, dependency lint started failing on the `ubuntu-latest` and `ember-beta` steps for all our PRs. We can add a config file for it, but it's strange that it's not failing for the other tests or that it just started failing.

File: package.json (hunk 4 above)
Comment (lukemelia) [reply package.json]: I saw that you removed it. If you do want to use it, I'm happy to help figure out how to get the css-element-queries module to work.

File: addon/components/boxel/input/index.hbs [hunk 5]
```diff
@@ -1,8 +1,9 @@
 <Input
   class="boxel-input"
   @id={{@id}}
-  @value={{@value}}
+  @value={{this.value}}
```
Comment (lukemelia) [inline addon/components/boxel/input/index.hbs]: Why don't we want to just rely on data-down/actions-up for this and require @onInput to update the passed in @value?

File: package.json (hunk 4 above)
Comment (burieberry) [reply package.json]: Ok we can make a separate ticket for this

File: addon/components/boxel/input/index.hbs (hunk 5 above)
Comment (burieberry) [reply addon/components/boxel/input/index.hbs]: Yes, that was the whole point. I got tired.

**Review by lukemelia — APPROVED** [review]

**Comment by burieberry** [comment]
@lukemelia FYI, I don't know if you've seen it but I've also added a TextField component which is a pre-packaged version of Boxel::Field wrapper with text type Boxel::Input. I'm thinking of doing the same for other form field types. If having that wrapper Field component doesn't work out or if it becomes unnecessary we can go back to not doing it that way.

**Comment by burieberry** [comment]
Ah also I'll double check the disabled etc state styling with Gardella during design review, just so that I don't delay merging this any longer.

**Pull request merged**